### PR TITLE
Cache facts and upload to foreman when they change

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Tests ok here.

### CODE REVIEW

File: extra/discover_host [hunk 1]
```diff
@@ -45,35 +47,47 @@ def discover_by_dns
...
-puts "Importing facts"
-begin
-  uri = URI.parse(discover_server)
-  http = Net::HTTP.new(uri.host, uri.port)
-  if uri.scheme == 'https' then
-    http.use_ssl = true
-    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+# loop, but only upload on changes
+while true do
+  uninteresting_facts=/kernel|operatingsystem|osfamily|ruby|path|time|swap|free|filesystem|version/i
```
Comment (ohadlevy) [inline extra/discover_host:85]: i would also look at selinux etc

File: extra/discover_host [hunk 2]
```diff
@@ -45,35 +47,47 @@ def discover_by_dns
...
+  facts = Facter.to_hash.reject! {|k,v| k =~ uninteresting_facts }
+  unless YAML.load(read_cache) == facts
```
Comment (ohadlevy) [inline extra/discover_host:87]: hashes are not always sorted, you might need to do full compare or sort it etc

File: extra/discover_host (hunk 1 above)
Comment (GregSutcliffe) [inline extra/discover_host:85]: Will add selinux to the main PR in a sec. From my testing, this cache didn't change in ~20 mins.

File: extra/discover_host (hunk 2 above)
Comment (GregSutcliffe) [inline extra/discover_host:87]: I tested it, it compares properly when unsorted. From the Hash docs:

Equality—Two hashes are equal if they each contain the same number of keys and if each key-value pair is equal to (according to Object#==) the corresponding elements in the other hash.

**Comment by GregSutcliffe** [comment]
Merging into #13

**Pull request closed without merging**